crates/oxc_linter/src/snapshots/no_multi_str.snap

```diff
@@ -0,0 +1,32 @@
+---
+source: crates/oxc_linter/src/tester.rs
+---
+  ⚠ eslint(no-multi-str): .
+   ╭─[no_multi_str.tsx:1:9]
+ 1 │ ╭─▶ var x = 'Line 1 \
```

Comment:
Is it possible only to point out on the `\` ?

Reply:
I suppose we can do that by `source.find('\')` and then creating a new Span with the StringLiteral.span.start + the index?

Reply:
LGTM

Reply:
I like the proposed `matches!` but would like to combine finding the character's position and match into one. It seems that `find` has [some issues](https://stackoverflow.com/a/46943949) with non-ASCII characters. So I went for finding the position, I assume that is as fast as `any`.

Not sure if I handled the `unwrap` well, but it seems impossible for the first position to be 0 and testing finds it will never be true:

```
  × eslint(no-multi-str): Unexpected multi string.
   ╭─[foo.js:7:12]
 6 │
 7 │ const b = "\
   ·            ─
 8 │ wjeifwe";
   ╰────
```

Reply:
I've added an extra test for this case.

Reply:
Our `start` is the `byte` offset from the start of the source text, I think `find` is ok here 🤔

https://github.com/oxc-project/oxc/blob/051ceb65396ea089f3213875afc8130112cb599d/crates/oxc_parser/src/lexer/mod.rs#L294-L295